mayastor/src/descriptor.rs

```diff
@@ -88,103 +88,52 @@ pub enum DmaError {
     ReadFailed { offset: u64, len: usize },
 }
 
-/// DmaBuf that is allocated from the memory pool
+/// Block Device Descriptor
 #[derive(Debug)]
-pub struct DmaBuf {
-    /// a raw pointer to the buffer
-    pub buf: *mut c_void,
-    /// the length of the allocated buffer
-    len: usize,
+pub struct Descriptor {
```

Comment:
@jkryl, Thanks for clean this up, in retrospect, the initially chosen naming here chose very poorly.

The initial idea was to use the descriptor -- that is analogous to a file descriptor, and it would provide all the methods to work with it. 

However, as we are moving towards multiple cores, there is an inherent problem with this approach that I'm hitting while implementing the new per segment rebuild implementation.

To sum that up, the idea here is that we divide the block device into segments, and have the nexus dispatch segments to whatever cores are available.

Because of that, that's where _this_ descriptor model brakes as per core we need an io channel. So it works only for whatever core we happen to initialise on. 

So we need to decouple the descriptor and the io_channel. We can then also get rid of the Rc around it. 

BTW, I'm not expecting you to do that as part of this PR or anything like that. I'm implementing two new types around these primitives.  I'll go back and update this part to make use of that once we agree on the code that implements it.

Reply:
thanks for explaining 👍